src/grammar/grammar.y

```diff
@@ -500,42 +510,46 @@ variable    :   identifier optElements
                     delete $1;
                     delete $2;
                 }
-            |   fxnCall '[' expression ']' optElements
+            |   fxnCall elementList
                 {
 #ifdef DEBUG_BISON_FLEX
                     printf("Parser inserting variable (FUNCTION_VAR) in syntax tree\n");
 #endif
-                    $$ = new SyntaxIndexOperation($1,$3);
-                    for (std::list<SyntaxElement*>::iterator it=$5->begin(); it!=$5->end(); ++it)
+                    $$ = $1;
+                    for (auto& element: *$2)
                     {
-                        $$ = new SyntaxIndexOperation($$,*it);
+                        $$ = new SyntaxIndexOperation($$, element);
                     }
-                    delete $5;
+                    delete $2;
                 }
-            |   '(' expression ')' '[' expression ']' optElements
+            |   '(' expression ')' elementList

```

Comment:
I couldn't directly see where elementList was defined. Is that now a more generic way of stitching together multiple "calls" within the same expression? Would 'rBDP(10,rootAge=1,lambda=1,taxa=[taxon("A"),taxon("B"),taxon("C"),taxon("D")])[3].rootAge()' work? (I couldn't come up with any better test.)

Reply:
elementList isn't new - its defined like this:

```
elementList :   '[' expression ']'              { $$ = new std::list<SyntaxElement*>(1, $2); }
            |   '[' ']'                         { $$ = new std::list<SyntaxElement*>(); }
            |   elementList '[' expression ']'  { $1->push_back($3); $$ = $1; }
            |   elementList '[' ']'             { $1->push_back( NULL ); $$ = $1; }
            ;
```

And optElements is defined like this:

```
optElements :   /* empty */                     { $$ = new std::list<SyntaxElement*>(); }
            |   elementList                     { $$ = $1; }
            ;
```

So, if we replace `'[' expression ']' optElements` with `elementList` there is _almost_ no change.  The only change is that now something like `f(x)[]` will parse.  Previously `f(x)[3][]` would parse but `f(x)[]` would not parse.

Possibly we should make a non_empty_elementList that looks like this:
```
nonEmptyElementList :   '[' expression ']'              { $$ = new std::list<SyntaxElement*>(1, $2); }
            |   elementList '[' expression ']'  { $1->push_back($3); $$ = $1; }
            ;
```
but I would leave that for a future cleanup.

Reply:
BTW, `rBDP(10,rootAge=1,lambda=1,taxa=[taxon("A"),taxon("B"),taxon("C"),taxon("D")])[3].rootAge()` does work.